server/src/test/java/com/hedera/block/server/persistence/storage/read/BlockAsLocalFileReaderTest.java

```diff
@@ -29,25 +51,64 @@
  * Tests for the {@link BlockAsLocalFileReader} class.
  */
 class BlockAsLocalFileReaderTest {
+    private BlockAsLocalFileWriter blockAsLocalFileWriterMock;
+    private BlockAsLocalFilePathResolver blockAsLocalFileResolverMock;
+    private BlockNodeContext blockNodeContext;
+    private PersistenceStorageConfig testConfig;
     private BlockAsLocalFileReader toTest;
 
+    @TempDir
+    private Path testLiveRootPath;
+
     @BeforeEach
-    void setUp() {
-        toTest = BlockAsLocalFileReader.newInstance();
+    void setUp() throws IOException {
+        blockNodeContext = TestConfigUtil.getTestBlockNodeContext(
+                Map.of(PERSISTENCE_STORAGE_LIVE_ROOT_PATH_KEY, testLiveRootPath.toString()));
+        testConfig = blockNodeContext.configuration().getConfigData(PersistenceStorageConfig.class);
+
+        final String testConfigLiveRootPath = testConfig.liveRootPath();
+        assertThat(testConfigLiveRootPath).isEqualTo(testLiveRootPath.toString());
+
+        blockAsLocalFileResolverMock = spy(BlockAsLocalFilePathResolver.of(testLiveRootPath));
+        blockAsLocalFileWriterMock = spy(BlockAsLocalFileWriter.of(blockNodeContext, blockAsLocalFileResolverMock));
+        toTest = BlockAsLocalFileReader.of(blockAsLocalFileResolverMock);
     }
 
     /**
      * This test aims to verify that the
      * {@link BlockAsLocalFileReader#read(long)} correctly reads a block with
      * a given block number.
-     *
-     * @param toRead parameterized, block number
      */
-    @ParameterizedTest
-    @MethodSource("validBlockNumbers")
-    void testSuccessfulBlockRead(final long toRead) {
-        // todo currently throws UnsupportedOperationException
-        assertThatExceptionOfType(UnsupportedOperationException.class).isThrownBy(() -> toTest.read(toRead));
+    @Test
```

Comment:
Why remove the parameterized test case source?  This runs fewer scenarios and reduces the number of requirements validated...

Reply:
The proper setup if I have kept the parameterized test was too difficult to do. I do actually write to the fs here, so it requires writing the test data first and then reading it. I'll see how to improve this with more test cases. Also, I will make another test to assert that an empty optional is returned if the block file does not exist (but block number is valid)

Reply:
Should have been relatively straightforward, mostly the test you have, but the block number is a parameter (and that determines the file name).
It will write a file, but it should also remove that file at the end of the test method, and in between it is basically unchanged...

##### General note
The use of parameterized tests is encouraged for _all_ unit tests; if the unit test is hard to parameterize, it probably needs to be rethought (not always, but in most cases, it can be parameterized with less change than you might expect).

Reply:
✅ I have streamlined this, the parameterized tests are back and I have added another helper method to generate block items for a block with specific number to facilitate this.